### PR TITLE
Fixed width multiplier

### LINKED ISSUE #973: Mobilenet v2 width multiplier incorrect
There seems to be a small issue with the width multiplier in mobilenet v2.  The official implementation rounds filter channels to a multiple of 8.  

For example, mobilenet v2 width 1.4.  The first conv layer has 44 channels as opposed to 48 in the official implementation:
model = torchvision.models.mobilenet_v2(width_mult=1.4)

for module in model.modules():
    if isinstance(module, nn.Conv2d):
        print(module.weight.shape)

torch.Size([44, 3, 3, 3])
torch.Size([44, 1, 3, 3])
torch.Size([22, 44, 1, 1])
torch.Size([132, 22, 1, 1])
torch.Size([132, 1, 3, 3])
torch.Size([33, 132, 1, 1])

Corresponding tensorflow 2.0 keras code:
model = tf.keras.applications.MobileNetV2(
    weights="imagenet", input_shape=(224, 224, 3), alpha=1.4)
model.summary()

__________________________________________________________________________________________________
Layer (type)                    Output Shape         Param #     Connected to                     
==================================================================================================
input_1 (InputLayer)            [(None, 224, 224, 3) 0                                            
__________________________________________________________________________________________________
Conv1_pad (ZeroPadding2D)       (None, 225, 225, 3)  0           input_1[0][0]                    
__________________________________________________________________________________________________
Conv1 (Conv2D)                  (None, 112, 112, 48) 1296        Conv1_pad[0][0]                  
__________________________________________________________________________________________________
bn_Conv1 (BatchNormalization)   (None, 112, 112, 48) 192         Conv1[0][0]                      
__________________________________________________________________________________________________
Conv1_relu (ReLU)               (None, 112, 112, 48) 0           bn_Conv1[0][0]                   
__________________________________________________________________________________________________
expanded_conv_depthwise (Depthw (None, 112, 112, 48) 432         Conv1_relu[0][0]                 
__________________________________________________________________________________________________
expanded_conv_depthwise_BN (Bat (None, 112, 112, 48) 192         expanded_conv_depthwise[0][0]    
__________________________________________________________________________________________________
expanded_conv_depthwise_relu (R (None, 112, 112, 48) 0           expanded_conv_depthwise_BN[0][0] 
__________________________________________________________________________________________________
expanded_conv_project (Conv2D)  (None, 112, 112, 24) 1152        expanded_conv_depthwise_relu[0][0
__________________________________________________________________________________________________
expanded_conv_project_BN (Batch (None, 112, 112, 24) 96          expanded_conv_project[0][0]      
__________________________________________________________________________________________________
block_1_expand (Conv2D)         (None, 112, 112, 144 3456        expanded_conv_project_BN[0][0]   
__________________________________________________________________________________________________
block_1_expand_BN (BatchNormali (None, 112, 112, 144 576         block_1_expand[0][0]             
__________________________________________________________________________________________________
block_1_expand_relu (ReLU)      (None, 112, 112, 144 0           block_1_expand_BN[0][0]          
__________________________________________________________________________________________________
block_1_pad (ZeroPadding2D)     (None, 113, 113, 144 0           block_1_expand_relu[0][0]        
__________________________________________________________________________________________________
block_1_depthwise (DepthwiseCon (None, 56, 56, 144)  1296        block_1_pad[0][0]                
__________________________________________________________________________________________________
block_1_depthwise_BN (BatchNorm (None, 56, 56, 144)  576         block_1_depthwise[0][0]          
__________________________________________________________________________________________________
block_1_depthwise_relu (ReLU)   (None, 56, 56, 144)  0           block_1_depthwise_BN[0][0]       
__________________________________________________________________________________________________
block_1_project (Conv2D)        (None, 56, 56, 32)   4608        block_1_depthwise_relu[0][0]     
__________________________________________________________________________________________________

I've implemented a fix here:
https://github.com/yaysummeriscoming/vision/blob/master/torchvision/models/mobilenet.py

Can I merge it in?

### PR DESCRIPTION
**Pull request by yaysummeriscoming**
Layer channels are now rounded to a multiple of 8, as per the official tensorflow implementation.  I found this fix when looking through: https://github.com/d-li14/mobilenetv2.pytorch

Fixes #973

### CODE REVIEW

File: torchvision/models/mobilenet.py [hunk 1]
```diff
@@ -74,12 +94,12 @@ def __init__(self, num_classes=1000, width_mult=1.0, inverted_residual_setting=N
                              "or a 4-element list, got {}".format(inverted_residual_setting))
 
         # building first layer
-        input_channel = int(input_channel * width_mult)
-        self.last_channel = int(last_channel * max(1.0, width_mult))
+        input_channel = _make_divisible(input_channel * width_mult, 8)
```
Comment (fmassa) [inline torchvision/models/mobilenet.py]: Should we add the `8` as a parameter, potentially letting the user choose different values?
I believe the only reason they used `8` here was because the underlying kernels might be optimized for multiples of 8

**Comment by yaysummeriscoming** [comment]
Good point on the divisor, I'll make that an option.  

Regarding the default value, I would suggest to keep it at 8?  Many people including myself are using this for benchmarking new algorithms, for which it's necessary to have this implementation 100% as per the official tensorflow implementation.  Only problem with this is pretrained models.  I guess I can convert these from the official tensorflow implementation, otherwise I have a training recipe that should provide slightly better results.

**Comment by fmassa** [comment]
@yaysummeriscoming for pre-trained models, can we use the scripts from `references/classification`? I used those scripts to training those models, and I should share the command-line arguments that I used to train `mobilenet`

**Comment by yaysummeriscoming** [comment]
@fmassa you were able to get up to par mobilenet results using that script, inc stepped LR?  If so I'm quite envious, I had to use some tricks to get up to scratch results.  I still think there's an argument to use the tensorflow weights however?  I forget where but there was some commotion about keras & pytorch models having slightly different accuracy levels.

In any case, pls share the command line arguments :)

**Comment by fmassa** [comment]
Yes, the model that I released uses that script.
Here are the arguments I used:

```
--model mobilenet_v2 --epochs 300 --lr 0.045 --wd 0.00004 --lr-step-size 1 --lr-gamma 0.98
```
With the following results

Top 1 | Top 5
| --- | --- |
71.878 | 90.286

**Comment by fmassa** [comment]
@yaysummeriscoming could you expose the `8` as a parameter to be passed? We can discuss about which default it should have, but I think it is important to expose it (at least for some backwards-compatibility)

**Comment by yaysummeriscoming** [comment]
Apologies for the delay, just put the change in.

**Comment by fmassa** [comment]
@yaysummeriscoming there is a linter error
```
./torchvision/models/mobilenet.py:152:1: W293 blank line contains whitespace
```
could you fix it?

**Comment by meijieru** [comment]
@fmassa Have you reproduce the result with width_multiplier 1.4 using that script?

**Comment by fmassa** [comment]
@meijieru I haven't tried any trainings other than the `width_multiplier=1.0` yet, and won't probably have time before August.
